uefi-test-runner/src/proto/ata/pass_thru.rs

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use uefi::boot;
+use uefi::boot::{OpenProtocolAttributes, OpenProtocolParams};
+use uefi::proto::ata::pass_thru::AtaPassThru;
+use uefi::proto::ata::AtaRequestBuilder;
+
+pub fn test() {
+    info!("Running ATA PassThru tests");
+
+    assert!(try_find_drive());
+}
+
+fn try_find_drive() -> bool {
+    let ata_ctrl_handles = boot::find_handles::<AtaPassThru>().unwrap();
+    assert_eq!(ata_ctrl_handles.len(), 1);
+
+    for handle in ata_ctrl_handles {
+        let params = OpenProtocolParams {
+            handle,
+            agent: boot::image_handle(),
+            controller: None,
+        };
+        let ata_pt = unsafe {
+            // don't open exclusive! That would break other tests
+            boot::open_protocol::<AtaPassThru>(params, OpenProtocolAttributes::GetProtocol).unwrap()
+        };
+        for mut device in ata_pt.iter_devices() {
+            // ATA IDENTIFY command
+            let request = AtaRequestBuilder::read_udma(ata_pt.io_align(), 0xEC)
+                .unwrap()
+                .with_timeout(core::time::Duration::from_millis(500))
+                .with_read_buffer(255)
+                .unwrap()
+                .build();
+            if let Ok(result) = device.execute_command(request) {
+                let bfr = result.read_buffer().unwrap();
+                // what the...
```

Comment:
? 😀

Reply:
I'm guessing this is reading the serial number? A clearer comment would be helpful

Reply:
Weird. I thought I removed that comment already. :thinking: 
Yeah, this is reading the serial number. It's in big endian :sweat: